internal/services/network/virtual_network_peering_resource.go

```diff
@@ -226,7 +226,7 @@ func retryVnetPeeringsClientCreateUpdate(d *pluginsdk.ResourceData, resGroup str
 		ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
 		defer cancel()
 
-		future, err := vnetPeeringsClient.CreateOrUpdate(ctx, resGroup, vnetName, name, peer)
+		future, err := vnetPeeringsClient.CreateOrUpdate(ctx, resGroup, vnetName, name, peer, network.SyncRemoteAddressSpaceTrue)
```

Comment:
This is the only available option for now, so decided not to make it into a parameter for now.